spec/index.html

```diff
@@ -7,36 +7,292 @@
+        </th>
+        <th>
+          Meaning
+        </th>
+      </tr>
+      <tr>
+        <td>
+          [[Set]]
+        </td>
+        <td>
+          an Object.
+        </td>
+        <td>
+          the Set or similar object.
+        </td>
+      </tr>
+      <tr>
+        <td>
+          [[Size]]
+        </td>
+        <td>
+          a non-negative integer or +∞
+        </td>
+        <td>
+          The reported size of the object.
+        </td>
+      </tr>
+      <tr>
+        <td>
+          [[Has]]
+        </td>
+        <td>
+          a function object
+        </td>
+        <td>
+          The `has` method of the object.
+        </td>
+      </tr>
+      <tr>
+        <td>
+          [[Keys]]
+        </td>
+        <td>
+          a function object
+        </td>
+        <td>
+          The `keys` method of the object.
+        </td>
+      </tr>
+    </table>
+  </emu-table>
+</emu-clause>
+
+<emu-clause id="sec-getsetrecord" type="abstract operation">
+  <h1>
+    GetSetRecord (
+      _obj_: an ECMAScript language value,
+    ): either a normal completion containing a Set Record or a throw completion
+  </h1>
+  <dl class="header">
+  </dl>
+  <emu-alg>
+    1. If _obj_ is not an Object, throw a *TypeError* exception.
+    1. Let _rawSize_ be ? Get(_obj_, *"size"*).
+    1. Let _numSize_ be ? ToNumber(_rawSize_).
+    1. NOTE: If _rawSize_ is *undefined*, then _numSize_ will be *NaN*.
+    1. If _numSize_ is *NaN*, throw a *TypeError* exception.
+    1. Let _intSize_ be ! ToIntegerOrInfinity(_numSize_).
+    1. Let _has_ be ? Get(_obj_, *"has"*).
+    1. If IsCallable(_has_) is *false*, throw a *TypeError* exception.
+    1. Let _keys_ be ? Get(_obj_, *"keys"*).
+    1. If IsCallable(_keys_) is *false*, throw a *TypeError* exception.
+    1. Return a new Set Record { [[Set]]: _obj_, [[Size]]: _intSize_, [[Has]]: _has_, [[Keys]]: _keys_ }.
+  </emu-alg>
+</emu-clause>
+
+<emu-clause id="sec-getkeysiterator" type="abstract operation">
+  <h1>
+    GetKeysIterator (
+      _setRec_: a Set Record,
+    ): either a normal completion containing an Iterator Record or a throw completion
+  </h1>
+  <dl class="header">
+  </dl>
+  <emu-alg>
+    1. Let _keysIter_ be ? Call(_setRec_.[[Keys]], _setRec_.[[Set]]).
+    1. If _keysIter_ is not an Object, throw a *TypeError* exception.
+    1. Let _nextMethod_ be ? Get(_keysIter_, *"next"*).
+    1. If IsCallable(_nextMethod_) is *false*, throw a *TypeError* exception.
+    1. Return a new Iterator Record { [[Iterator]]: _keysIter_, [[NextMethod]]: _nextMethod_, [[Done]]: *false* }.
+  </emu-alg>
+</emu-clause>
+
+<emu-clause id="sec-setdatahas" type="abstract operation">
+  <h1>
+    SetDataHas (
+      _resultSetData_: a List of either ~empty~ or ECMAScript language values,
+      _value_: an ECMAScript language value,
+    ): a boolean
+  </h1>
+  <dl class="header">
+  </dl>
+  <emu-alg>
+    1. If _value_ is *-0*<sub>𝔽</sub>, set _value_ to *+0*<sub>𝔽</sub>.
```

Comment:
what's the benefit of this line, since SameValueZero is used below?

Reply:
Ah, good catch, that's left over from a previous factoring. Will remove.